File: src/procedures/LightningProcedure.java
package procedures;

import led.ColorRGB;
import led.LEDDataBundle;

public class LightningProcedure extends Procedure {
	
	private float mSpeed;
	private float mAtenuation;
	
	public LightningProcedure(LEDDataBundle _bundle) {
		super(_bundle);
		
		mSpeed = _bundle.speed;
		mAtenuation = _bundle.value1;
		mSteps = _bundle.duration * 50;
	}

	@Override
	public void start() {
		
	}
	
	@Override
	void update() {
		mStrip.setAllPixels(ColorRGB.white50.dim((float)(Math.abs(Math.sin(mStep / mSpeed)))).toSystemColor());
		
		mSpeed += mAtenuation;
		
		if (mStep > mSteps) {
			finishProcedure(true);
		}
		
		mStep++;
	}
}

File: src/main/java/procedures/LightningProcedure.kt
package procedures

import com.github.iamniklas.liocore.led.LEDDataBundle
import com.github.iamniklas.liocore.led.colorspace.ColorRGB
import com.github.iamniklas.liocore.procedures.Procedure

class LightningProcedure(_bundle: LEDDataBundle) : Procedure(_bundle) {
    private var mSpeed: Float
    private val mAtenuation: Float
    override fun start() {}
    public override fun update() {
        strip!!.setAllPixels(
            ColorRGB.white50.dim(Math.abs(Math.sin((step / mSpeed).toDouble())).toFloat()).toSystemColor()
        )
        mSpeed += mAtenuation
        if (step > steps) {
            finishProcedure(true)
        }
        step++
    }

    init {
        mSpeed = _bundle.speed!!
        mAtenuation = _bundle.value1!!
        steps = _bundle.duration!! * 50
    }
}